packages/api/src/derives/staking/stakingAccount.ts

```diff
@@ -0,0 +1,101 @@
+import {ApiInterfaceRx} from '@cennznet/api/types';
+import {DerivedStakingQuery} from '@polkadot/api-derive/types';
+import {memo} from '@polkadot/api-derive/util';
+import {createType, Option, TypeRegistry, Vec} from '@polkadot/types';
+import {
+  AccountId,
+  Exposure,
+  Keys,
+  Nominations,
+  RewardDestination,
+  StakingLedger,
+  ValidatorPrefs,
+} from '@polkadot/types/interfaces';
+import {ITuple} from '@polkadot/types/types';
+import {combineLatest, Observable, of} from 'rxjs';
+import {map, switchMap} from 'rxjs/operators';
+
+type MultiResultV2 = [
+  Option<AccountId>,
+  Option<Nominations>,
+  RewardDestination,
+  Exposure,
+  [ValidatorPrefs],
+  Option<Keys>,
+  Option<StakingLedger>
+];
+
+function unwrapSessionIds(
+  stashId: AccountId,
+  queuedKeys: [AccountId, Keys][],
+  nextKeys: Option<Keys>
+): {nextSessionIds: AccountId[]; sessionIds: AccountId[]} {
+  let sessionIds: AccountId[] = [];
+  const idKeys = queuedKeys.find(([currentId]): boolean => currentId.eq(stashId));
+  if (idKeys) {
+    sessionIds = idKeys[1];
+  }
+  const nextSessionIds = nextKeys.unwrapOr([] as AccountId[]);
+
+  return {
+    nextSessionIds,
+    sessionIds,
+  };
+}
+
+function retrieve(
+  api: ApiInterfaceRx,
+  stashId: AccountId
+): Observable<[Vec<ITuple<[AccountId, Keys]>>, MultiResultV2]> {
+  return combineLatest([
+    api.query.session.queuedKeys<Vec<ITuple<[AccountId, Keys]>>>(),
```

Comment:
Not sure if the queued session keys are of the nature close enough to be combined with staking details in the same query.

Reply:
I would suggest having a different query for session keys if we need them.

Reply:
DerivedStakingQuery type is expected to return following types
export interface DerivedStakingQuery extends DerivedStakingStash {
    accountId: AccountId;
    nextSessionIds: AccountId[];
    sessionIds: AccountId[];
    stakingLedger?: StakingLedger;
}
and  DerivedStakingStash
export interface DerivedStakingStash {
    controllerId?: AccountId;
    nominators?: AccountId[];
    nominateAt?: EraIndex;
    rewardDestination?: RewardDestination;
    nextKeys?: Keys;
    stakers?: Exposure;
    stashId?: AccountId;
    validatorPrefs?: ValidatorPrefs;
}
These are types from polkadot and are used to get all the relevant details at one go..

Reply:
Yup I get that. What I mean is you can do a query for getting the staking details and then another query for session ids. The third step is refining session ids to find the ones related to us and the final step is putting the relevant info next to each other to form DerivedStakingQuery or DerivedStakingStash.

Reply:
updated with two different queries